src/OpenTelemetry.Extensions/Trace/PartialActivityProcessor.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Logs;
+
+namespace OpenTelemetry.Extensions.Trace;
+
+/// <summary>
+/// Activity processor that exports LogRecord on Activity start, during it's lifetime as a heartbeat and on end.
+/// </summary>
+/// <remarks>
+/// Add this processor *before* exporter related Activity processors.
+/// </remarks>
+public class PartialActivityProcessor : BaseProcessor<Activity>
+{
+    private readonly ManualResetEvent shutdownTrigger;
+    private readonly BaseExporter<LogRecord> logExporter;
+    private readonly BaseProcessor<LogRecord> logProcessor;
+    private readonly ILogger<PartialActivityProcessor> logger;
+    private ILoggerFactory loggerFactory;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PartialActivityProcessor"/> class.
+    /// </summary>
+    /// <param name="logExporter">Log exporter to be used.</param>
+    public PartialActivityProcessor(BaseExporter<LogRecord> logExporter)
+    {
+#if NET
+        ArgumentNullException.ThrowIfNull(logExporter);
+#else
+        if (logExporter == null)
+        {
+            throw new ArgumentOutOfRangeException(nameof(logExporter));
+        }
+#endif
+        this.logExporter = logExporter;
+        this.logProcessor = new SimpleLogRecordExportProcessor(logExporter);
```

Comment:
I believe it's generally preferred to use a [BatchLogRecordExportProcessor](https://github.com/open-telemetry/opentelemetry-dotnet/blob/8c1e63894f0286b1c48b0448a866fdb3bd603fe8/src/OpenTelemetry/Logs/Processor/BatchLogRecordExportProcessor.cs) over a simple one processor for efficiency.

Reply:
i used simple log record export processor because i'd like to have logs pushed asap. do you think i can achieve the same with batch log record export processor?

Reply:
Yes, the batch processor still sends fairly quickly (the default [batch window](https://github.com/open-telemetry/opentelemetry-dotnet/blob/8c1e63894f0286b1c48b0448a866fdb3bd603fe8/src/OpenTelemetry/BatchExportProcessor.cs#L18) is 5s but could be configured lower if needed). It means the SDK doesn't need to establish and send each log record one at a time, instead it waits a short period and sends a group together.

To avoid having to expose those settings in this extension, you could take the processor as an argument so someone can choose how to configure it. Eg use a simple processor, change batch times, etc. I'd recommend the default be the Batch processor if one isn't provided.